### PR TITLE
Adding random adapter

### PR DESCRIPTION
**Pull request by RLeclair**
Adding random adapter (RANDOM_seed and RANDOM_stub) to improve randomness calls for uws when using a system supporting POSIX.

### CODE REVIEW

File: configs/azure_iot_build_rules.cmake [hunk 1]
```diff
@@ -74,7 +74,7 @@ elseif(UNIX) #LINUX OR APPLE
     set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Werror")
     set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Werror")
     if(NOT (IN_OPENWRT OR APPLE))
-        set (CMAKE_C_FLAGS "-D_POSIX_C_SOURCE=200112L ${CMAKE_C_FLAGS}")
+        set (CMAKE_C_FLAGS "-D_POSIX_C_SOURCE=200112L -D_XOPEN_SOURCE=500 ${CMAKE_C_FLAGS}")
```
Comment (ewertons) [inline configs/azure_iot_build_rules.cmake:77]: Sorry, could you add this comment on top please?
// _XOPEN_SOURCE=500 is required for glibc to expose random and srandom.

File: devdoc/uws_frame_encoder_requirements.md [hunk 2]
```diff
@@ -64,7 +64,7 @@ extern int uws_frame_encoder_encode(BUFFER_HANDLE encode_buffer, WS_FRAME_TYPE o
 
 **SRS_UWS_FRAME_ENCODER_01_052: [** If `reserved` has any bits set except the lowest 3 then `uws_frame_encoder_encode` shall fail and return NULL. **]**
 
-**SRS_UWS_FRAME_ENCODER_01_053: [** In order to obtain a 32 bit value for masking, `gb_rand` shall be used 4 times (for each byte). **]**
+**SRS_UWS_FRAME_ENCODER_01_053: [** In order to obtain a 32 bit value for masking, `RANDOM_generate` shall be used 4 times (for each byte). **]**
```
Comment (ewertons) [inline devdoc/uws_frame_encoder_requirements.md:67]: Excellent

**Review by ewertons — APPROVED** [review]
Excellent example of how to add an adapter to c-utility. Thank you very much!

File: configs/azure_iot_build_rules.cmake (hunk 1 above)
Comment (RLeclair) [reply configs/azure_iot_build_rules.cmake:77]: Added comment! Thanks for the review @ewertons!

**Pull request merged**